pkgs/applications/editors/notepad-next/default.nix

```diff
@@ -0,0 +1,36 @@
+{ mkDerivation, lib, fetchFromGitHub, qmake, libsForQt5 }:
+
+mkDerivation rec {
+  pname = "notepad-next";
+  version = "0.5.1";
+
+  src = fetchFromGitHub {
+    owner = "dail8859";
+    repo = "NotepadNext";
+    rev = "v${version}";
+    sha256 = "sha256-J7Ngt6YtAAZsza2lN0d1lX3T8gNJHp60sCwwaLMGBHQ=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ qmake libsForQt5.qt5.qttools ];
+  qmakeFlags = [ "src/NotepadNext.pro" ];
+
+  # References
+  #  https://github.com/dail8859/NotepadNext/blob/master/doc/Building.md
+  #  https://github.com/dail8859/NotepadNext/issues/99
+  postPatch = ''
+    substituteInPlace ./src/NotepadNext/NotepadNext.pro --replace /usr $out
+  '';
+
+  CXXFLAGS = "-std=gnu++1z";
+
+  enableParallelBuilding = true;
+
+  meta = with lib; {
+    homepage = "https://github.com/dail8859/NotepadNext";
+    description = "Notepad++-like editor for the Linux desktop";
+    license = licenses.gpl3;
```

Comment:
`gpl3` is unclear, use `gpl3Plus` or `gpl3Only`. See https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206 for more details.